### PR TITLE
Add authorization to organizations endpoint

### PR DESCRIPTION
**Pull request by rudigiesler**
This first PR will probably have a lot of groundwork for permissions handling, so it will just be adding the permissions for the organizations endpoint. Adding the specific permissions to the other endpoints can happen in separate PRs.

### CODE REVIEW

**Comment by rudigiesler** [comment]
@justinvdm Ready for review.

File: authapi/permissions.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    '''
+    def __init__(self, permission_type):
+        self.permission_type = permission_type
+
+    def has_permission(self, permission, request, view):
+        user = request.user
+        permissions = get_user_permissions(user)
+        permissions = find_permission(permissions, self.permission_type)
+        if permissions.count() >= 1:
+            return True
```
Comment (justinvdm) [inline authapi/permissions.py]: should we return `False` otherwise?

File: authapi/permissions.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class AllowObjectPermission(AllowPermission):
+    '''This component checks whether a user has a specific permission type,
+    and that the object id matches the permission object id.'''
+    def has_object_permission(self, permission, request, view, obj):
+        user = request.user
+        permissions = get_user_permissions(user)
+        permissions = find_permission(
+            permissions, self.permission_type, str(obj.pk))
+        if permissions.count() >= 1:
+            return True
```
Comment (justinvdm) [inline authapi/permissions.py]: Same here about returning `False`.

File: authapi/permissions.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    def object_permission_set(self):
+        '''
+        All users can read. admins, org:admins, and users with org:write
+        permission for the specific organization can update. admins and
+        org:admins can create.
+        '''
+        return Or(
+            AllowOnlySafeHttpMethod,
+            AllowAdmin,
+            And(Or(AllowCreate, AllowModify), AllowPermission('org:admin')),
```
Comment (justinvdm) [inline authapi/permissions.py]: Should `org:admin` also be an `AllowObjectPermission`?

File: authapi/tests/test_organizations.py [hunk 4]
```diff
@@ -211,6 +235,153 @@ def test_summary_serializer(self):
...
+        SeedPermission.objects.all().delete()
+        self.add_permission(user, 'org:write', org1.pk)
+        response = self.client.put(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Authenticated request, org:admin permissions
+        SeedPermission.objects.all().delete()
+        self.add_permission(user, 'org:admin')
+        response = self.client.put(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (justinvdm) [inline authapi/tests/test_organizations.py]: Do we need a test here for wrong org for `org:admin`? Similarly for other methods' tests.

**Comment by justinvdm** [comment]
Left a few minor comments. Looks good!

File: authapi/permissions.py (hunk 3 above)
Comment (rudigiesler) [inline authapi/permissions.py]: Ah yes, I misread the docs, only admins can create organizations, `org:admin` gives other permissions adding users and teams to their org. `org:admin` is tied to a specific organization, I'll change it.

**Comment by rudigiesler** [comment]
@justinvdm I think I've addressed your comments. Ready for another review.

**Comment by justinvdm** [comment]
:+1: !

**Pull request merged**